### DISCUSSION TITLE
click issue

### DISCUSSION THREAD

**Discussion by hooke007** [tomasklaen/uosc#931] [General]
Clicking on chapter indicators will also run the commands bind with `MBTN_LEFT`

@christoph-heinrich Any ideas?

**Comment by christoph-heinrich** [comment]
The problem is that chapters listen to the `primary_down` event, but not to the `primary_up`, which leaves the `up` event unhandled, and thus it gets forwarded to the mouse click.

The cheap way out is to listen to the up event once a chapter has been clicked and do nothing with it. All other cases of `primary_down` event handlers also happen to listen to `primary_up` (either in general, or once it has been clicked), which is why it's only problematic for chapters.

It would be somewhat nicer to keep track of if the down event has been handled and then ignore the up event if no handler for it has been registered, but I'm not sure if that would maybe lead to problems in some way somewhere???
I can't think of a specific scenario where this would be problematic, it just seems like the kind of magic where something doesn't end up getting forwarded that actually should have been.

@tomasklaen what do you think?

I'll make a PR for listening to `primary_up` after clicking and ignoring it.

**Reply by tomasklaen** [reply]
We have `primary_click` now, so stuff like this should be switched to it. I probably just missed this one.

Unless there's a reason why chapters have to be on up/down. Can't recall any atm.

**Reply by christoph-heinrich** [reply]
I totally forgot about the `_click` variant. That works fine too.
